docs/arch/adr-27.md

```diff
@@ -0,0 +1,53 @@
+# ADR 27: Subject Viewer Config
+
+Created: June 26, 2020
+
+## Context
+
+A workflow determines which subject viewer to use with the `workflow.configuration.subject_viewer` property. Subject viewers include `singleImage`, `lightCurve`, `multiFrame`, and `subjectGroup` at creation of this ADR. Some subject viewers can utilize, or require, additional configuration information.
+
+### [multiFrame](https://github.com/zooniverse/front-end-monorepo/tree/master/packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/MultiFrameViewer)
+
+A workflow using the multi-frame subject viewer might have a preference regarding:
+marks per frame: some might prefer marks filtered per frame, like a transcription workflow where each frame represents a unique page to transcribe with marks only relevant to each page, while other workflows might prefer marks persist between frames [, like ?] (`multi_image_clone_markers` in PFE)
+pan and zoom: some might prefer pan and zoom reset per frame, like a transcription workflow where each frame represents a unique page to transcribe, while other workflows might prefer pan and zoom maintained between frames, like [Wildcam Gorongosa](https://www.zooniverse.org/projects/zooniverse/wildcam-gorongosa/classify) or [Backyard Worlds](https://www.zooniverse.org/projects/marckuchner/backyard-worlds-planet-9/classify) (in flipbook mode, not separate frames)
+
+### [subjectGroup](https://github.com/zooniverse/front-end-monorepo/tree/master/packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SubjectGroupViewer)
+
+A workflow using the subject-group subject viewer might want to define the subject cell width, height, or style, or the subject viewer grid columns or grid rows.
+
+## Decision
+
+Subject viewer configuration is an object stored in `workflow.configuration.subject_viewer_config` and is structured as follows for the subject viewers noted:
+
+`multiFrame`:
+
+```javascript
+{
+  filter_marks_per_frame: [boolean]
+  maintain_pan_zoom: [boolean]
```

Comment:
Originally had something like `pan_zoom: ['maintain', 'reset']`, but thought might be better to have as boolean if makes sense?

Not including rotate in the key, but technically I think we would include, so open to suggestion if ok as is or should also note `rotation` in key.

Reply:
I could see an enumerable being useful if it's possible to have any other states... Just a note on the code example, maybe use < and > instead of brackets since this sort of looks like you mean an array that contains a boolean?

Reply:
We should note here in the decision that `filter_marks_per_frame` is replacing `multi_image_clone_markers`.

Reply:
> Not including rotate in the key, but technically I think we would include, so open to suggestion if ok as is or should also note rotation in key.

The setting is more than maintain the rotation between frames though? Isn't the scale and pan position maintained as well or no?

Reply:
That's correct, in Wildcam Gorongosa rotation, scale, and pan position are maintained between frames. Maybe `pan_zoom` could be renamed to something like `positioning` and include pan, zoom, and rotation?

Reply:
I wonder if there will ever be a request to maintain only one of these? I was thinking about suggesting this be an enumerable if various values, but then we would have to support each of these configurations and that might be too complex and not actually wanted. I would support making the proposed key `maintain_positioning` though